### PR TITLE
RavenDB-18695 - add a new license type

### PR DESCRIPTION
**Pull request by grisha-kotler**

### CODE REVIEW

File: src/Raven.Server/Commercial/LicenseType.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@ public enum LicenseType
 
         Community = 1,
 
-        Trial = 2,
+        Essential = 2,
```
Comment (ayende) [inline src/Raven.Server/Commercial/LicenseType.cs]: I don't think this is correct. 
We don't want to _replace_ `Trial` (although that isn't used at all for us).
We need a new value (and changing `Trial => Reserved`)

Another thing to consider is the backward compatibility here, no?

Comment (grisha-kotler) [reply src/Raven.Server/Commercial/LicenseType.cs]: done

File: src/Raven.Server/Commercial/LicenseType.cs [hunk 2]
```diff
@@ -8,12 +8,14 @@ public enum LicenseType
...
+        Reserved = 2,
 
         Professional = 3,
 
         Enterprise = 4,
 
-        Developer = 5
+        Developer = 5,
+
+        Essential = 6
```
Comment (ayende) [inline src/Raven.Server/Commercial/LicenseType.cs:19]: how would existing code handle this new license type?
IIRC, we aren't actually doing anything with it, right?

Comment (grisha-kotler) [reply src/Raven.Server/Commercial/LicenseType.cs:19]: Anything that cannot be recognized will be `Community`.
We use the `Developer` type to check the certificate expiration.
The other usage is displaying it in the studio.

**Pull request merged**